Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -111,42 +102,33 @@ TextureCacheBase::~TextureCacheBase()
 
 void TextureCacheBase::OnConfigChanged(VideoConfig& config)
 {
-  if (g_texture_cache)
+  if (config.bHiresTextures != backup_config.hires_textures ||
+      config.bCacheHiresTextures != backup_config.cache_hires_textures)
   {
-    if (config.bHiresTextures != backup_config.s_hires_textures ||
-        config.bCacheHiresTextures != backup_config.s_cache_hires_textures)
-    {
-      HiresTexture::Update();
-    }
+    HiresTexture::Update();
+  }
 
-    // TODO: Invalidating texcache is really stupid in some of these cases
-    if (config.iSafeTextureCache_ColorSamples != backup_config.s_colorsamples ||
-        config.bTexFmtOverlayEnable != backup_config.s_texfmt_overlay ||
-        config.bTexFmtOverlayCenter != backup_config.s_texfmt_overlay_center ||
-        config.bHiresTextures != backup_config.s_hires_textures)
-    {
-      g_texture_cache->Invalidate();
+  // TODO: Invalidating texcache is really stupid in some of these cases
+  if (config.iSafeTextureCache_ColorSamples != backup_config.color_samples ||
+      config.bTexFmtOverlayEnable != backup_config.texfmt_overlay ||
+      config.bTexFmtOverlayCenter != backup_config.texfmt_overlay_center ||
+      config.bHiresTextures != backup_config.hires_textures)
+  {
+    Invalidate();
 
-      TexDecoder_SetTexFmtOverlayOptions(g_ActiveConfig.bTexFmtOverlayEnable,
-                                         g_ActiveConfig.bTexFmtOverlayCenter);
-    }
+    TexDecoder_SetTexFmtOverlayOptions(g_ActiveConfig.bTexFmtOverlayEnable,
+                                       g_ActiveConfig.bTexFmtOverlayCenter);
+  }
 
-    if ((config.iStereoMode > 0) != backup_config.s_stereo_3d ||
-        config.bStereoEFBMonoDepth != backup_config.s_efb_mono_depth)
-    {
-      g_texture_cache->DeleteShaders();
-      if (!g_texture_cache->CompileShaders())
-        PanicAlert("Failed to recompile one or more texture conversion shaders.");
-    }
+  if ((config.iStereoMode > 0) != backup_config.stereo_3d ||
+      config.bStereoEFBMonoDepth != backup_config.efb_mono_depth)
+  {
+    g_texture_cache->DeleteShaders();
+    if (!g_texture_cache->CompileShaders())
+      PanicAlert("Failed to recompile one or more texture conversion shaders.");
   }
 
-  backup_config.s_colorsamples = config.iSafeTextureCache_ColorSamples;
-  backup_config.s_texfmt_overlay = config.bTexFmtOverlayEnable;
-  backup_config.s_texfmt_overlay_center = config.bTexFmtOverlayCenter;
-  backup_config.s_hires_textures = config.bHiresTextures;
-  backup_config.s_cache_hires_textures = config.bCacheHiresTextures;
-  backup_config.s_stereo_3d = config.iStereoMode > 0;
-  backup_config.s_efb_mono_depth = config.bStereoEFBMonoDepth;
+  SetBackupConfig(config);
 }
 
 void TextureCacheBase::Cleanup(int _frameCount)
```

Comment:
Would it be worth renaming _frameCount to something else here (due to starting with an underscore)?